packages/lit-dev-content/src/global/hydrate-common-components.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const hydrateOnIdle = () => {
+  import('../components/lazy-svg.js');
+  import('../components/litdev-version-selector.js');
+};
+
+if ('requestIdleCallback' in window) {
+  (window as any).requestIdleCallback(hydrateOnIdle);
+} else {
+  // Safari doesn't have idle callback so just wait 200ms
+  setTimeout(hydrateOnIdle, 200);
+}
+
+const searches = Array.from(document.body.querySelectorAll('litdev-search'));
+
+for (const search of searches) {
+  search.addEventListener(
+    'focus',
+    async () => {
+      await import('../components/litdev-search.js');
+
+      search.focus();
```

Comment:
On a slow network. If someone starts typing, and then the element upgrades. It seems the input is cleared. Is there anyway to preserve the input during hydration?

Repro. Throttle network. Refresh page. Click search and start typing. On upgrade the input is cleared.
Expected behavior, the input is not cleared and maybe the suggestions appear.

https://user-images.githubusercontent.com/15080861/170516948-2b07da1e-5be7-478f-81d8-4751e9cb5a54.mov

Reply:
it's because of the `live` directive. When it hydrates, it will render `''` in the textfield. I also found another small issue for when there are two of these on the page and they hydrate. the static nature of the loading is causing funkyness